<?xml version="1.0" ?>
<launch>

  <!-- args -->
  <arg name="robot_name" default=""/>  <!-- "a_bot", "b_bot", or "c_bot" -->
  <arg name="camera_name" default=""/>
  <arg name="eye_on_hand" default="false"/>
  <arg name="config" default="real"/> <!-- "sim", "real", "osaka", or "aist" -->

	<!-- Start cad matching -->
	<include file="$(find o2as_cad_matching)/launch/search_server.launch"/>
	<node name="data_collection_server" pkg="omron_cad_matching" type="data_collection_server" output="screen"/>
	<node name="reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>

  <!-- Start robot controller -->
  <include file="$(find o2as_calibration)/launch/robot_control.launch">
    <arg name="config" value="$(arg config)"/>
  </include>

  <!-- Start calibrator -->
  <node name="hand_eye_solver" pkg="visp_hand2eye_calibration" type="visp_hand2eye_calibration_calibrator">
    <remap from="/compute_effector_camera_quick" to="compute_effector_camera_quick"/>
    <param name="eye_on_hand" type="string" value="$(arg eye_on_hand)"/>
  </node>

	<!-- Start calibration -->
  <!-- <arg name="robot_base_frame" default="$(arg robot_name)_base_link"/>
  <arg name="robot_effector_frame" default="$(arg robot_name)_ee_link"/>
  <arg name="image_dir" default="$(find o2as_calibration)/data/image"/>
  <node name="calibration" pkg="o2as_calibration" type="run_handeye_calibration.py" output="screen">
    <param name="image_dir" type="string" value="$(arg image_dir)"/>
    <param name="eye_on_hand" type="string" value="$(arg eye_on_hand)"/>
    <param name="robot_base_frame" type="string" value="$(arg robot_base_frame)"/>
    <param name="robot_effector_frame" type="string" value="$(arg robot_effector_frame)"/>
  </node> -->

</launch>
